core/src/main/scala/org/broadinstitute/dsde/rawls/expressions/parser/antlr/DataRepoInputExpressionValidationVisitor.scala

```diff
@@ -22,7 +25,32 @@ class DataRepoInputExpressionValidationVisitor(rootEntityType: Option[String],
   override def visitEntityLookup(ctx: EntityLookupContext): Try[Unit] = {
     rootEntityType match {
       case Some(rootTableName) => validateEntityLookup(rootTableName, ctx)
-      case None => Failure(new RawlsException("Expressions beginning with \"this.\" are only allowed when running with workspace data model. However, workspace attributes can be used."))
+      case None => Failure(new RawlsExceptionWithErrorReport(
+        ErrorReport(StatusCodes.BadRequest, "Expressions beginning with \"this.\" are only allowed when running with workspace data model. However, workspace attributes can be used.") // it seems like it would get here only if methodConfiguration.rootEntityType is not set, so this error message doesn't make sense to me. Shouldn't it be something like "Missing Root Entity Type. Please select one and try again."
+      ))
+    }
+  }
+
+  override def visitNamespace(ctx: TerraExpressionParser.NamespaceContext): Try[Unit] = {
+    rootEntityType match {
+      case Some(_) => {
+        val maybeParent = Option(ctx.parent.parent.getChild(0).getText)
+        maybeParent match {
+          case Some(maybeWorkspace) => {
+            if (maybeWorkspace == "workspace.")
+              super.visitNamespace(ctx)
+            else
+              Failure(new RawlsExceptionWithErrorReport(
+                ErrorReport(StatusCodes.BadRequest, "Expressions with \"namespace: name\" are not valid for BigQuery")
+              ))
+          }
+          case None => Failure(new RawlsExceptionWithErrorReport(
+            ErrorReport(StatusCodes.BadRequest, "Expressions with \"namespace: name\" are not valid for BigQuery")
+          ))
+        }
+
+      }
+      case None => super.visitNamespace(ctx)
```

Comment:
sorry, I gave you bad advice... I forgot we need to allow namespaces on workspace expressions.

The better place to do this is in `validateEntityLookup` where you can check `entityLookupContext.attributeName().namespace()`.

The code that looks like `ctx.parent.parent.getChild(0).getText` is the part that is concerning to me here. Also the checking for "workspace.". Both feel like brute force approaches that depend on the grammar structure but don't use the parseTree. If the grammar changes in the future these things may break and the compiler will miss it.